ghga_connector/core/api_calls.py

```diff
@@ -179,6 +311,51 @@ def download_api_call(
     return download_url, file_size, NO_RETRY_TIME
 
 
+def restart_multipart_upload(
+    api_url: str, file_id: str, error: Exception
+) -> Tuple[str, int]:
+    """Try to cancel the currently running multipart upload and start a new one"""
+
+    pending_upload = get_pending_uploads(api_url=api_url, file_id=file_id)
+    if pending_upload is None:
```

Comment:
> This only means that there is no currently pending upload, right? It doesn't mean that the user is not allowed to create a new upload.

Well, the restart-function is only called, after an attempt was made to start a download. I'll change the program flow.

> This might also help to come to a better separation between core and cli code.

I don't get what you mean by that.